### PR TITLE
[Keyboard] Add dnworks numpad

### PR DESCRIPTION
**Pull request by itsme-zeix**
## Description

Added support for dnworks numpad rev1. Thank you!
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dnworks/numpad/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+[0] = LAYOUT(
+		KC_ESC,  KC_APP,  KC_TAB,  KC_BSPC,
+    
+		KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS,
+		KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
+		KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
+		KC_P1,   KC_P2,   KC_P3,   KC_PENT,
+		KC_P0,   KC_P0,   KC_PDOT, KC_PENT
+	),
```
Comment (drashna) [inline keyboards/dnworks/numpad/keymaps/default/keymap.c:28]:
```suggestion
        KC_ESC,  KC_APP,  KC_TAB,  KC_BSPC,
    
        KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS,
        KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
        KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
        KC_P1,   KC_P2,   KC_P3,   KC_PENT,
        KC_P0,   KC_P0,   KC_PDOT, KC_PENT
    ),
```

File: keyboards/dnworks/numpad/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+[0] = LAYOUT(
+		KC_ESC,  KC_APP,  KC_TAB,  KC_BSPC,
+    
+		KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS,
+		KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
+		KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
+		KC_P1,   KC_P2,   KC_P3,   KC_PENT,
+		KC_P0,   KC_P0,   KC_PDOT, KC_PENT
+	),
```
Comment (drashna) [inline keyboards/dnworks/numpad/keymaps/via/keymap.c:28]:
```suggestion
[0] = LAYOUT_ortho_6x4(
        KC_ESC,  KC_APP,  KC_TAB,  KC_BSPC,
    
        KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS,
        KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
        KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
        KC_P1,   KC_P2,   KC_P3,   KC_PENT,
        KC_P0,   KC_P0,   KC_PDOT, KC_PENT
    ),
```

File: keyboards/dnworks/numpad/keymaps/default/keymap.c (hunk 1 above)
Comment (drashna) [inline keyboards/dnworks/numpad/keymaps/default/keymap.c:28]:
```suggestion
[0] = LAYOUT_ortho_6x4(
        KC_ESC,  KC_APP,  KC_TAB,  KC_BSPC,
    
        KC_NUM,  KC_PSLS, KC_PAST, KC_PMNS,
        KC_P7,   KC_P8,   KC_P9,   KC_PPLS,
        KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
        KC_P1,   KC_P2,   KC_P3,   KC_PENT,
        KC_P0,   KC_P0,   KC_PDOT, KC_PENT
    ),
```

File: keyboards/dnworks/numpad/info.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        "pid": "0x2937",
+        "device_version": "0.0.1"
+    },
+    "diode_direction": "COL2ROW",
+    "matrix_pins": {
+        "rows": ["GP11", "GP4", "GP3", "GP2", "GP1", "GP0", "GP24"],
+        "cols": ["GP5", "GP6", "GP7", "GP25"]
+    },
+    "layouts": {
+        "LAYOUT": {
```
Comment (drashna) [inline keyboards/dnworks/numpad/info.json:18]:
```suggestion
        "LAYOUT_ortho_6x4": {
```

File: keyboards/dnworks/numpad/info.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                { "matrix": [4, 0], "x": 0, "y": 4.25 },
+                { "matrix": [4, 1], "x": 1, "y": 4.25 },
+                { "matrix": [4, 2], "x": 2, "y": 4.25 },
+                { "matrix": [4, 3], "x": 3, "y": 4.25 },
+                { "matrix": [5, 0], "x": 0, "y": 5.25 },
+                { "matrix": [5, 1], "x": 1, "y": 5.25 },
+                { "matrix": [5, 2], "x": 2, "y": 5.25 },
+                { "matrix": [5, 3], "x": 3, "y": 5.25 }
+            ]
+        }
```
Comment (drashna) [inline keyboards/dnworks/numpad/info.json:45]:
```suggestion
        },
        "LAYOUT_numpad_6x4": {
            "layout": [
                { "matrix": [0, 0], "x": 0, "y": 0 },
                { "matrix": [0, 1], "x": 1, "y": 0 },
                { "matrix": [0, 2], "x": 2, "y": 0 },
                { "matrix": [0, 3], "x": 3, "y": 0 },
                { "matrix": [1, 0], "x": 0, "y": 1.25 },
                { "matrix": [1, 1], "x": 1, "y": 1.25 },
                { "matrix": [1, 2], "x": 2, "y": 1.25 },
                { "matrix": [6, 3], "x": 3, "y": 1.25 },
                { "matrix": [2, 0], "x": 0, "y": 2.25 },
                { "matrix": [2, 1], "x": 1, "y": 2.25 },
                { "matrix": [2, 2], "x": 2, "y": 2.25 },
                { "matrix": [3, 0], "x": 0, "y": 3.25 },
                { "matrix": [3, 1], "x": 1, "y": 3.25 },
                { "matrix": [3, 2], "x": 2, "y": 3.25 },
                { "matrix": [3, 3], "x": 3, "y": 2.25, "h": 2 },
                { "matrix": [4, 0], "x": 0, "y": 4.25 },
                { "matrix": [4, 1], "x": 1, "y": 4.25 },
                { "matrix": [4, 2], "x": 2, "y": 4.25 },
                { "matrix": [5, 0], "x": 0, "y": 5.25, "w": 2 },
                { "matrix": [5, 2], "x": 2, "y": 5.25 },
                { "matrix": [5, 3], "x": 3, "y": 4.25, "h": 2 }
            ]
        }
```

File: keyboards/dnworks/numpad/info.json [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        "vid": "0x4C23",
+        "pid": "0x2937",
+        "device_version": "0.0.1"
+    },
+    "diode_direction": "COL2ROW",
+    "matrix_pins": {
+        "rows": ["GP11", "GP4", "GP3", "GP2", "GP1", "GP0", "GP24"],
+        "cols": ["GP5", "GP6", "GP7", "GP25"]
+    },
+    "layouts": {
```
Comment (drashna) [inline keyboards/dnworks/numpad/info.json:17]:
```suggestion
    "community_layouts": [
        "numpad_6x4",
        "ortho_6x4"
    },
    "layouts": {
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by itsme-zeix** [comment]
@drashna Small bump for a review! Requested changes have been made.

**Review by drashna — APPROVED** [review]

File: keyboards/dnworks/numpad/info.json
[the remark this answers is not in the document]
Comment (itsme-zeix) [reply keyboards/dnworks/numpad/info.json:1]: Thanks for the review, change made as requested!

**Review by fauxpark — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**